docs/dummyvm/src/slots.rs

```diff
@@ -0,0 +1,73 @@
+use mmtk::util::constants::LOG_BYTES_IN_ADDRESS;
+use mmtk::{
+    util::{Address, ObjectReference},
+    vm::slot::{MemorySlice, SimpleSlot},
+};
+
+// A binding may implement their own type of slots.
+// See https://github.com/mmtk/mmtk-core/blob/master/src/vm/tests/mock_tests/mock_test_slots.rs for different kinds of slots.
+pub type DummyVMSlot = SimpleSlot;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct DummyVMMemorySlice(*mut [ObjectReference]);
```

Comment:
MemorySlice is an advanced topic.  Currently it is only useful for the `memory_region_copy` barrier (and can potentially be used by `Scanning::scan_object`, too. See [this issue](https://github.com/mmtk/mmtk-core/issues/986)).  The VM binding developer will only need it when they start to support generational GC, which, in my opinion, is the fourth step in VM binding development (after (1) supporting NoGC, (2) supporting a non-copying GC such as MarkSweep, and (3) supporting a copying non-generational GC such as SemiSpace).  That may be one year or two into the VM binding development.  At that time, the VM binding developers should have enough knowledge to implement its own `MemorySlice`.  The Ruby binding has been developed for more than two years, and is just about to support StickyImmix.  But it is currently using a write barrier in the form of "this object has been modified", so it still doesn't use `MemorySlice`.

We should simply define `DummyVMMemorySlice` as `mmtk::vm::slot::UnimplementedMemorySlice` in order not to clutter the boilerplate.

But we should push an implementation like this into `mmtk::vm::slot`, probably named `SimpleMemorySlice`, to be paired with `SimpleSlot`, and then deprecate the `impl MemorySlice for Range<Address>` just like `impl Slot for Address`.

Reply:
I removed the `slots` module.